### PR TITLE
[pioneeravr] Add Listening Mode Channels

### LINKED ISSUE #1653: [pioneeravr] Add setting and querying listening modes functionality
<a alt="bounty" href="https://www.bountysource.com/issues/40590803"></a>

It would be useful (to me) to add a bit of functionality for setting / querying listening modes:

Set Listening Mode:  [****SR<CR>]
Request Listening Mode: [?L<CR>]
Request Resolution Status: [?VTC<CR>]

Being able to send custom commands and be given the ACSII response would probably add this functionality for all use cases.

### PR DESCRIPTION
**Pull request by gersilex**
# [pioneeravr] Add Listening Mode Channels

This improves the *pioneeravr* binding by adding two channels for setting and reading the *Listening Mode* and reading the *Playing Listening Mode*.

## Listening Modes

The *Listening Mode* is set by user to instruct the AVR how to treat the audio signal and do upscaling, downscaling and amplification. This settings corresponds to the settings made with the remote control or front panel. What the AVR actually does with each setting/input-signal-combination can be read out using the *Playing Listening Mode* channel.

## Playing Listening Modes

The *Playing Listening Mode* is the Listening Mode that is actually playing as opposed to the *Listening Mode* set by the user. The *Playing Listening Mode* is what the display on the device shows.

## Forum Discussions

- [PioneerAVR 2 - Listen Mode](https://community.openhab.org/t/pioneeravr-2-listen-mode/18595)

## Related Issues

Closes #1653 

## JARs For Testing

https://github.com/gersilex/openhab2-addons/releases

### CODE REVIEW

File: bundles/org.openhab.binding.pioneeravr/src/main/resources/ESH-INF/thing/thing-types.xml [hunk 1]
```diff
@@ -198,6 +198,8 @@
 			<channel id="volumeDb" typeId="volumeChannelDb" />
 			<channel id="mute" typeId="muteChannel" />
 			<channel id="setInputSource" typeId="setInputSourceChannel" />
+			<channel id="ListeningMode" typeId="ListeningModeChannel" />
```
Comment (Hilbrand) [inline bundles/org.openhab.binding.pioneeravr/src/main/resources/ESH-INF/thing/thing-types.xml]: All other channels start with a lower case character:
```suggestion
			<channel id="listeningMode" typeId="ListeningModeChannel" />
```

Comment (gersilex) [reply bundles/org.openhab.binding.pioneeravr/src/main/resources/ESH-INF/thing/thing-types.xml]: You are totally right. Nice eyes, indeed!

Comment (Hilbrand) [reply bundles/org.openhab.binding.pioneeravr/src/main/resources/ESH-INF/thing/thing-types.xml]: You need to do it at all places where this channel and typeid is used. Not only here. I just put it here as reference :smile:

Comment (gersilex) [reply bundles/org.openhab.binding.pioneeravr/src/main/resources/ESH-INF/thing/thing-types.xml]: Done in https://github.com/openhab/openhab2-addons/pull/5515/commits/c3207e4c94bca224ae4759810280959207285614

**Comment by gersilex** [comment]
@Hilbrand I am waiting for two approved reviews and will then squash and reword my commits (including signing off) according to the contribution guidelines. I hope this is correct and you are not waiting for me to do it first.

**Comment by Hilbrand** [comment]
Not really waiting. I'll approve it. Looks fine. If you can already handle the signoff than the next maintainer can directly merge the pr if he approves it.

**Comment by gersilex** [comment]
@J-N-K https://github.com/openhab/openhab2-addons/pull/5515#pullrequestreview-232369548 :white_check_mark: Done.

**Comment by J-N-K** [comment]
Thanks. Looks good so far. After Jenkins finished, i‘ll merge.

**Comment by J-N-K** [comment]
FTR: Jenkins fails due to the null-bugs fixed by #5518

**Pull request merged**